src/components/Main.js

```diff
@@ -0,0 +1,105 @@
+import {nanoid} from 'nanoid';
+import {useState, useEffect} from 'react';
+
+import data from '../data.json';
+import StyledMain from '../style/StyledMain';
+
+export default function Form() {
+  const [selectedBodypart, setSelectedBodypart] = useState('');
+  const [plans, setPlans] = useState([]);
+  const [weeks, setWeeks] = useState(JSON.parse(localStorage.getItem('my_plan')) || []);
+  //handlesubmit
+  function handleSubmit(event) {
+    event.preventDefault();
+    saveWeek();
+  }
+  //setting the day workout
+  function createPlan(event) {
+    const newPlan = event.target.value;
+    setPlans([...plans, {name: newPlan, id: nanoid()}]);
+  }
+  // workouts store
+  useEffect(() => {
+    localStorage.setItem('my_plan', JSON.stringify(weeks));
+  }, [weeks]);
+  //setting the sets
+  function updateSets(id, value) {
+    setPlans([...plans.slice(0, id), {...plans[id], sets: value}, ...plans.slice(id + 1)]);
+  }
```

Comment:
Please rename these to `index` as it is the `index`:
```suggestion
  function updateSets(index, value) {
    setPlans([...plans.slice(0, index), {...plans[index], sets: value}, ...plans.slice(index + 1)]);
  }
```